LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -223,199 +232,229 @@ internal LoRaDevice(string devAddr, string devEUI, ILoRaDeviceClientConnectionMa
-                    {
-                        if (Enum.IsDefined(typeof(LoRaRegionType), loRaRegion))
-                        {
-                            this.region = new ChangeTrackingProperty<LoRaRegionType>(TwinProperty.Region, loRaRegion);
-                        }
-                    }
+            if (twin.Properties.Desired.Contains(TwinProperty.GatewayID))
+                GatewayID = twin.Properties.Desired[TwinProperty.GatewayID].Value as string;
 
-                    if (LoRaRegion == LoRaRegionType.NotSet)
-                    {
-                        this.logger.LogError($"invalid region value: {regionValue}");
-                    }
-                }
+            _ = UpdateIsOurDevice(configuration.GatewayID);
 
-                //  We are prioritizing the choice of the join channel from reported properties (set for OTAA devices)
-                //  over the manually provisioned channel (set in desired properties for ABP devices).
-                if (twin.Properties.Reported.Contains(TwinProperty.CN470JoinChannel))
-                {
-                    ReportedCN470JoinChannel = GetTwinPropertyIntValue(twin.Properties.Reported[TwinProperty.CN470JoinChannel].Value);
-                }
-                else if (twin.Properties.Desired.Contains(TwinProperty.CN470JoinChannel))
-                {
-                    DesiredCN470JoinChannel = GetTwinPropertyIntValue(twin.Properties.Desired[TwinProperty.CN470JoinChannel].Value);
-                }
+            if (twin.Properties.Desired.Contains(TwinProperty.SensorDecoder))
+                SensorDecoder = twin.Properties.Desired[TwinProperty.SensorDecoder].Value as string;
 
-                if (twin.Properties.Desired.Contains(TwinProperty.Supports32BitFCnt))
+            InitializeFrameCounters(twin);
+
+            if (twin.Properties.Desired.Contains(TwinProperty.DownlinkEnabled))
+            {
+                DownlinkEnabled = GetTwinPropertyBoolValue(twin.Properties.Desired[TwinProperty.DownlinkEnabled].Value);
+            }
+
+            if (twin.Properties.Desired.Contains(TwinProperty.PreferredWindow))
+            {
+                var preferredWindowTwinValue = GetTwinPropertyIntValue(twin.Properties.Desired[TwinProperty.PreferredWindow].Value);
+                if (preferredWindowTwinValue == Constants.ReceiveWindow2)
+                    PreferredWindow = preferredWindowTwinValue;
+            }
+
+            if (twin.Properties.Desired.Contains(TwinProperty.Deduplication))
+            {
+                var val = twin.Properties.Desired[TwinProperty.Deduplication].Value as string;
+                _ = Enum.TryParse<DeduplicationMode>(val, true, out var mode);
+                Deduplication = mode;
+            }
+
+            if (twin.Properties.Desired.Contains(TwinProperty.ClassType))
+            {
+                if (string.Equals("c", (string)twin.Properties.Desired[TwinProperty.ClassType], StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: Can we combine the two "if" in one?

Reply:
I know that those are all flagged by the tool now because the indention changed. I didn't plan to do a complete refactoring of all the code in the LoRaDevice as part of that change. There is going to be a push to get rid of all the dynamic type handling in there, we can do some more cleanup then.